File: java/src/backjoon/backjoon_8_1.java
package backjoon;

import java.util.Scanner;

public class backjoon_8_1 {

    public static void main(String[] args) {

        Scanner sc = new Scanner(System.in);
        long A = sc.nextLong(); // 1000
        long B = sc.nextLong(); // 70
        long C = sc.nextLong(); // 170

        int n = 1;

        // B가 C와 같으면, 손익분기점은 평생 넘을수없음
        if(B>=C) {
            System.out.println("-1");
            return;
        }

        long minus = A;
        long plus = 0 ;

        while(true){

            minus += B ;
            plus += C ;

            if(plus > minus) {
                System.out.println(n);
                break;
            }
            else {
                n++;
            }
        }
    }
}

File: kotlin/src/backjoon/kotlin/backjoon_8_1.kt
package backjoon.kotlin

fun main(args: Array<String>) {

    var str = readLine().toString()
    var A = str.split(" ").get(0).toLong()// 1000
    var B = str.split(" ").get(1).toLong() // 70
    var C = str.split(" ").get(2).toLong() // 170

    // B가 C랑 같거나 클경우, 절대 손익분기점 못넘음
    if(B >= C) {
        println("-1")
    }

    var minus:Long = A // 손해가 나는 금액
    var plus:Long = 0 // 이득이 나는 금액
    var n:Int = 1 // 손익분기점에 도달할 횟수

    while(true) {
        minus += B
        plus += C

        if(plus > minus) {
            println(n)
            break;
        }
        else {
            n++;
        }
    }

}

